lstchain/statistics.py

```diff
@@ -66,3 +67,56 @@ def var(self):
     def std(self):
         '''Get the current sample std. dev. of all tracked statistics'''
         return np.sqrt(self.var)
+
+
+
+def sigma_clipped_mean_std(values, axis=0, max_sigma=4, n_iterations=5):
+    '''
+    Compute robust estimates of mean and std via sigma clipping
+
+    Values outside max_sigma * std are removed from the sample and then
+    mean and std are computed again.
+    '''
+
+
+    # support masked array
+    if hasattr(values, 'mask'):
+        values = values.copy()
+    else:
+        mask = np.zeros(values.shape, dtype=bool)
+        values = np.ma.array(values, mask=mask, copy=True)
+
+    original_mask = values.mask.copy()
+    mean = values.mean(axis=axis)
+    std = values.std(axis=axis)
+
+    for _ in range(n_iterations):
+        values.mask = original_mask | (np.abs(values - mean) >= (max_sigma * std))
```

Comment:
Hi @maxnoe,
we should not here add "original_mask = values.mask " ?

Reply:
yes, actually I did not mean that (the original mask is a copy isn't it?), what I meant is that we should keep track in the loop of the masked pixels in the iteration because they are other wise lost in the next one and again re-detected in the successive one and so on.

Reply:
It is ok only to output at the end the final mask

Reply:
>  what I meant is that we should keep track in the loop of the masked pixels in the iteration because they are other wise lost in the next one and again re-detected in the successive one and so on.

That's exactly the reason why the original mask is `or` ed with the currently detected outliers. Since both mean and std are updated in the loop, it is ok to recompute what are outliers and combine it with the previously excluded pixels.

Reply:
ok